### PR TITLE
Add formulas and references to metrics docs

### PR DESCRIPTION
**Pull request by teddykoker**
See [docs build artifact](https://github.com/PyTorchLightning/pytorch-lightning/suites/1549672639/artifacts/27676361).

### CODE REVIEW

File: pytorch_lightning/metrics/classification/precision_recall.py [hunk 1]
```diff
@@ -25,12 +25,15 @@
 
 
 class Precision(Metric):
-    """
-    Computes the precision metric.
+    r"""
+    Computes `Precision <https://en.wikipedia.org/wiki/Precision_and_recall>`_:
+
+    .. math:: \text{Precision} = \frac{TP}{TP + FP}
```
Comment (awaelchli) [inline pytorch_lightning/metrics/classification/precision_recall.py:31]: I would also make TP and FP text

**Review by awaelchli — APPROVED** [review]
love this!

File: pytorch_lightning/metrics/classification/confusion_matrix.py [hunk 2]
```diff
@@ -24,9 +24,11 @@
...
 class ConfusionMatrix(Metric):
     """
-    Computes the confusion matrix. Works with binary, multiclass, and multilabel data.
-    Accepts logits from a model output or integer class values in prediction.
-    Works with multi-dimensional preds and target.
+    Computes the `confusion matrix
+    <https://en.wikipedia.org/wiki/Confusion_matrix>`_.  Works with binary,
+    multiclass, and multilabel data.  Accepts logits from a model output or
+    integer class values in prediction.  Works with multi-dimensional preds and
+    target.
```
Comment (s-rog) [inline pytorch_lightning/metrics/classification/confusion_matrix.py:31]: An issue came up before about how the confusion matrix metric can't be directly logged like the other metrics since it outputs an array, should we add a reminder here as well?

File: pytorch_lightning/metrics/classification/confusion_matrix.py [hunk 3]
```diff
@@ -24,9 +24,11 @@
 
 class ConfusionMatrix(Metric):
     """
-    Computes the confusion matrix. Works with binary, multiclass, and multilabel data.
-    Accepts logits from a model output or integer class values in prediction.
-    Works with multi-dimensional preds and target.
+    Computes the `confusion matrix
+    <https://en.wikipedia.org/wiki/Confusion_matrix>`_.  Works with binary,
```
Comment (SkafteNicki) [inline pytorch_lightning/metrics/classification/confusion_matrix.py:28]: We cannot use link to wikipedia for confusion matrix because they describe it as preds=rows and true=columns where we follow sklearns convention preds=columns and true=rows. Instead we should either link to sklearn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html#sklearn.metrics.confusion_matrix or rewrite the text ourself

**Review by SkafteNicki — COMMENTED** [review]
`Accuracy` metric could probably also use a update. Not that I think this wiki page is any good https://en.wikipedia.org/wiki/Accuracy_and_precision so maybe just state formula

File: pytorch_lightning/metrics/classification/confusion_matrix.py (hunk 2 above)
Comment (justusschock) [reply pytorch_lightning/metrics/classification/confusion_matrix.py:31]: we could also add a plot function and the log this as an image?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/metrics/classification/confusion_matrix.py (hunk 3 above)
Comment (teddykoker) [reply pytorch_lightning/metrics/classification/confusion_matrix.py:28]: Will do!

**Review by SkafteNicki — APPROVED** [review]
